### PR TITLE
Metrics Infrastructure

### PR DESCRIPTION
**Pull request by amir-blox**
Resolve https://github.com/bloxapp/ssv/issues/159

* Added infra for metrics collection
* Enable [pprof](https://github.com/google/pprof)
* Added collectors for validators, ibft and process related metrics

### CODE REVIEW

File: cli/operator/node.go [hunk 1]
```diff
@@ -113,6 +115,13 @@ var StartNodeCmd = &cobra.Command{
 		if err := operatorNode.StartEth1(eth1.HexStringToSyncOffset(cfg.ETH1SyncOffset)); err != nil {
 			Logger.Fatal("failed to start eth1", zap.Error(err))
 		}
+		go func() {
+			metricsHandler := metrics.NewMetricsHandler(Logger)
+			if err := metricsHandler.Start(http.NewServeMux(), ":15000"); err != nil {
```
Comment (nivBlox) [inline cli/operator/node.go:120]: lets add port to config

File: metrics/collect.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+)
+
+// Collector interface that represents a specific collector
+type Collector interface {
+	Collect() ([]string, error)
+	ID() string
+}
+
+var (
+	prefix        = "ssv"
```
Comment (nivBlox) [inline metrics/collect.go:15]: need to be distinct name - e.g "ssv-collect"

**Review by nivBlox — CHANGES_REQUESTED** [review]

File: cli/operator/node.go [hunk 3]
```diff
@@ -24,6 +27,8 @@ type config struct {
 	SSVOptions                 operator.Options `yaml:"ssv"`
 	ETH1Options                eth1.Options     `yaml:"eth1"`
 
+	Metrics           string `yaml:"Metrics" env:"METRICS" env-default:"" env-description:"comma seperated list of metrics collectors"`
+	MetricsAPIAddr    string `yaml:"MetricsAPIAddr" env:"METRICS_API_ADDR" env-default:":15000" env-description:"address of metrics api"`
```
Comment (lior-blox) [inline cli/operator/node.go:31]: @amir-blox lets change it to MetricsPort

File: cli/operator/node.go [hunk 4]
```diff
@@ -24,6 +27,8 @@ type config struct {
 	SSVOptions                 operator.Options `yaml:"ssv"`
 	ETH1Options                eth1.Options     `yaml:"eth1"`
 
+	Metrics           string `yaml:"Metrics" env:"METRICS" env-default:"" env-description:"comma seperated list of metrics collectors"`
```
Comment (lior-blox) [inline cli/operator/node.go:30]: @amir-blox empty means all?

File: cli/operator/node.go [hunk 5]
```diff
@@ -116,6 +121,7 @@ var StartNodeCmd = &cobra.Command{
 		if err := operatorNode.StartEth1(eth1.HexStringToSyncOffset(cfg.ETH1Options.ETH1SyncOffset)); err != nil {
 			Logger.Fatal("failed to start eth1", zap.Error(err))
 		}
+		go startMetricsHandler(Logger, cfg)
```
Comment (lior-blox) [inline cli/operator/node.go:124]: @amir-blox lets add a way to disable metrics, could be that if port is not provided(default ) don't collect metrics

**Pull request merged**